### ISSUE TITLE
Meaning and function of red line in dev mode at the top of the screen is not clear

### ISSUE THREAD

**Issue by andrii-i** [jupyterlab/jupyterlab#14008]
## Description

Meaning and function of red line at the top of the screen shown in dev mode is not clear. 
Communication by color only is not clear and is not accessible. 

## Reproduce

1. Launch jupyterlab in dev mode
2. Note 4px high red border at the top of the screen

![image](https://user-images.githubusercontent.com/26686070/218905476-d8f86e11-a391-4dc9-bdba-ea7ed74c8bc3.png)

## Expected behavior

All jupyterlab UI elements have clear function and are accessible, red dev mode line has accessible indication of its function. For example, `[ dev mode ]` text displayed as a part of the red line

## Context

Operating System and version: MacOS 12.6.3
Browser and version: Chrome version 109.0.5414.87 (Official Build) (arm64)
JupyterLab version: head-of-master (WIP 4.0.0)

**Comment by jtpio** [comment]
> For example, `[ dev mode ]` text displayed as a part of the red line

Adding this text sounds good. Although it would be great to not increase the height of the red line and keep it as small as possible.

**Comment by krassowski** [comment]
Could it be displayed on hover by adding `title` attribute?

**Comment by andrii-i** [comment]
> Could it be displayed on hover by adding `title` attribute?

I think this would solve a problem of line having no indication of its function while not increasing its size / keeping it as small as possible as @jtpio suggests. 

Currently red line is added at css level so it needs to be reimplement as element with `title` attribute.

**Comment by GabrielaVives** [comment]
The only problem I see with the text displayed on hover over the red line is that it may not be clear for users with different kinds of color blindness? For example, if it is perceived as grey, it may not look any different than the background color and users will not even think to hover over this area to see what it means.

Could something like this work? (it could be a different UI style, I just gave it a guess)
<img width="1282" alt="image" src="https://user-images.githubusercontent.com/105775162/228002745-f1cdf467-dbd8-4374-886b-124b94710f15.png">
